src/contracts/examples/fastlane-online/OuterHelpers.sol

```diff
@@ -72,13 +89,26 @@ contract OuterHelpers is FastLaneOnlineInner {
         userOp = _getUserOperation(swapper, swapIntent, baselineCall, deadline, gas, maxFeePerGas);
     }
 
-    function makeThogardsWifeHappy() external onlyAsControl withUserLock {
+    function makeThogardsWifeHappy() external onlyAsControl withUserLock(msg.sender) {
         require(msg.sender == CARDANO_ENGINEER_THERAPY_FUND, "ERR - NOT MAD JUST DISAPPOINTED");
         uint256 _rake = rake;
         rake = 0;
         SafeTransferLib.safeTransferETH(CARDANO_ENGINEER_THERAPY_FUND, _rake);
     }
 
+    function _simulateSolverOp(
+        UserOperation calldata userOp,
+        SolverOperation calldata solverOp
+    )
+        internal
+        returns (bool valid)
+    {
+        DAppOperation memory _dAppOp = _getDAppOp(solverOp.userOpHash, userOp.deadline);
+
+        // NOTE: Valid is false when the solver fails even if postOps is successful
+        (valid,,) = ISimulator(SIMULATOR).simSolverCall(userOp, solverOp, _dAppOp);
```

Comment:
This might be an issue. If `tokenUserSells` is the native token, then we would need to send enough with `msg.value` along with this sim call for an accurate simulation of the solverOp.

But this would require the solver to send enough native token when calling `addSolverOp()` to pay for their `congestionBuyIn` + the user's `amountUserSells`.

We can refund the `amountUserSells` of native token to the solver after the sim happens, but this is probably not great for solver UX, and requires some extra accounting logic to separate the two pieces making up `msg.value`.

Reply:
Good catch. I think this is part of a larger issue that exists with the simulator around passing msg.value when we use the simulator onchain. 

I'm wondering if we should add a baseline amount of value to the simulator so that we don't keep running into this issue with other modules.

Reply:
Thats probably a good idea.